diff --git a/Makefile b/Makefile
index becb95317..74c74eade 100644
--- a/Makefile
+++ b/Makefile
@@ -73,7 +73,7 @@ GITREVDATE=$(shell git log -n 1 --format="%cd" --date=format:%Y%m%d-%H%M%S)
 # Reduces build time and binary sizes considerably.
 # That's only needed if you use gdb or nm.
 # If you need that, build manually without these flags.
-GOFLAGS := "-ldflags=-s -w -X github.com/google/syzkaller/prog.GitRevision=$(REV) -X 'github.com/google/syzkaller/prog.gitRevisionDate=$(GITREVDATE)'"
+GOFLAGS := -gcflags "all=-N -l" -ldflags "-X github.com/google/syzkaller/prog.GitRevision=$(REV) -X 'github.com/google/syzkaller/prog.gitRevisionDate=$(GITREVDATE)'"
 
 GOHOSTFLAGS ?= $(GOFLAGS)
 GOTARGETFLAGS ?= $(GOFLAGS)
diff --git a/sys/targets/targets.go b/sys/targets/targets.go
index a623dbfb6..8bb69d4d9 100644
--- a/sys/targets/targets.go
+++ b/sys/targets/targets.go
@@ -578,7 +578,8 @@ var oses = map[string]osCommon{
 
 var (
 	commonCFlags = []string{
-		"-O2",
+		"-g",
+		"-O0",
 		"-pthread",
 		"-Wall",
 		"-Werror",
