diff --git a/pkg/report/freebsd.go b/pkg/report/freebsd.go
index 37c50eaf8b..c7512a6e42 100644
--- a/pkg/report/freebsd.go
+++ b/pkg/report/freebsd.go
@@ -4,7 +4,13 @@
 package report
 
 import (
+	"bytes"
+	"fmt"
+	"path/filepath"
 	"regexp"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
 )
 
 type freebsd struct {
@@ -26,7 +32,54 @@ func (ctx *freebsd) Parse(output []byte) *Report {
 	return simpleLineParser(output, freebsdOopses, freebsdStackParams, ctx.ignores)
 }
 
+// dirty implementation, but at least it works
 func (ctx *freebsd) Symbolize(rep *Report) error {
+	kernelBin := filepath.Join(ctx.kernelDirs.Obj, ctx.target.KernelObject)
+	re := regexp.MustCompile(`#[0-9]+ 0xffffffff`)
+	lines := bytes.Split(rep.Report, []byte("\n"))
+
+	// collect all addresses from the stack trace
+	type pair struct {
+		lnum int
+		addr []byte
+	}
+	var pairs []pair
+	for i, line := range lines {
+		if !re.Match(line) {
+			continue
+		}
+		parts := bytes.Split(line, []byte(" "))
+		if len(parts) < 2 {
+			continue
+		}
+		pairs = append(pairs, pair{lnum: i, addr: parts[1]})
+	}
+
+	// symbolize addresses in batch
+	timeout := 10 * time.Second
+	args := []string{"-e", kernelBin}
+	for _, pair := range pairs {
+		args = append(args, string(pair.addr))
+	}
+	output, err := osutil.RunCmd(timeout, "", "addr2line", args...)
+	if err != nil {
+		return err
+	}
+	outputSlices := bytes.Split(output, []byte("\n"))
+	outputSlices = outputSlices[:len(outputSlices)-1] // remove trailing empty line
+	if len(outputSlices) != len(pairs) {
+		return fmt.Errorf("len(outputSlices)=%v, want %v", len(outputSlices), len(pairs))
+	}
+
+	// append symbolized lines to the end of each line of report
+	for i, pair := range pairs {
+		line := lines[pair.lnum]
+		symbolized := outputSlices[i]
+		symbPath := osutil.Abs(string(symbolized))
+		lines[pair.lnum] = fmt.Appendf(nil, "%s %s", line, symbPath)
+	}
+	rep.Report = bytes.Join(lines, []byte("\r\n"))
+
 	return nil
 }
 
diff --git a/sys/syz-extract/freebsd.go b/sys/syz-extract/freebsd.go
index 1bcd9823f8..1279ffccb1 100644
--- a/sys/syz-extract/freebsd.go
+++ b/sys/syz-extract/freebsd.go
@@ -18,6 +18,7 @@ func (*freebsd) prepare(sourcedir string, build bool, arches []*Arch) error {
 	if !build {
 		return fmt.Errorf("freebsd requires -build flag")
 	}
+	// run make depend for GENERIC config?
 	return nil
 }
 
@@ -52,6 +53,7 @@ func (*freebsd) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]ui
 		"-D__BSD_VISIBLE=1",
 		"-DCOMPAT_FREEBSD13",
 		"-DCOMPAT_FREEBSD14",
+		"-DCOMPAT_43TTY",
 		"-I", filepath.Join(arch.sourceDir, "sys"),
 		"-I", filepath.Join(arch.sourceDir, "sys", "sys"),
 		"-I", filepath.Join(arch.sourceDir, "sys", "contrib", "ck", "include"),
diff --git a/tools/syz-symbolize/symbolize.go b/tools/syz-symbolize/symbolize.go
index 0a884119b8..7d01f765b1 100644
--- a/tools/syz-symbolize/symbolize.go
+++ b/tools/syz-symbolize/symbolize.go
@@ -4,6 +4,7 @@
 package main
 
 import (
+	"bytes"
 	"flag"
 	"fmt"
 	"os"
@@ -54,6 +55,7 @@ func main() {
 		tool.Failf("failed to create reporter: %v", err)
 	}
 	text, err := os.ReadFile(flag.Args()[0])
+	text = bytes.Replace(text, []byte("\r\n"), []byte("\n"), -1)
 	if err != nil {
 		tool.Failf("failed to open input file: %v", err)
 	}
