diff --git a/prog/size.go b/prog/size.go
index d354aadbea..320eb20d41 100644
--- a/prog/size.go
+++ b/prog/size.go
@@ -205,7 +205,7 @@ func (r *randGen) mutateSize(arg *ConstArg, parent []Arg, fields []Field) bool {
 						if targetType.Elem.Varlen() {
 							return false
 						}
-						elemSize = targetType.Elem.Size()
+						elemSize = max(targetType.Elem.Size(), 1)
 					}
 				}
 				break
diff --git a/sys/syz-extract/linux.go b/sys/syz-extract/linux.go
index 8b365a2068..dca6cf00b6 100644
--- a/sys/syz-extract/linux.go
+++ b/sys/syz-extract/linux.go
@@ -133,6 +133,7 @@ func (*linux) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]uint
 		"-O3", // required to get expected values for some __builtin_constant_p
 		"-I.",
 		"-D__KERNEL__",
+		"-D__DVB_CORE__",
 		"-DKBUILD_MODNAME=\"-\"",
 		"-I" + sourceDir + "/arch/" + headerArch + "/include",
 		"-I" + buildDir + "/arch/" + headerArch + "/include/generated/uapi",
