diff --git a/vm/qemu/qemu.go b/vm/qemu/qemu.go
index d5d10d66bf..20e27d0de6 100644
--- a/vm/qemu/qemu.go
+++ b/vm/qemu/qemu.go
@@ -227,6 +227,12 @@ var archConfigs = map[string]*archConfig{
 		RngDev:                 "virtio-rng-pci",
 		UseNewQemuImageOptions: true,
 	},
+	"openbsd/amd64": {
+		Qemu:     "qemu-system-x86_64",
+		QemuArgs: "-enable-kvm -cpu host,migratable=off",
+		NetDev:   "virtio-net-pci",
+		RngDev:   "virtio-rng-pci",
+	},
 	"darwin/amd64": {
 		Qemu: "qemu-system-x86_64",
 		QemuArgs: strings.Join([]string{
